code/modules/mining/equipment_locker.dm

```diff
@@ -500,38 +517,108 @@
 	origin_tech = "bluespace=2"
 	slot_flags = SLOT_BELT
 
-/obj/item/device/wormhole_jaunter/attack_self(mob/user)
+	var/obj/item/device/radio/belt_radio
+	var/radio_key = /obj/item/device/encryptionkey/headset_cargo
+	var/radio_channel = "Supply"
+
+	var/medical_sensors = FALSE
+	var/ticks_in_crit = 0
+	var/threshold = 2
+
+/obj/item/device/wormhole_lifebelt/New()
+	. = ..()
+	desc += desc_sensors
+	SSobj.processing += src
+
+	belt_radio = new /obj/item/device/radio(src)
+	belt_radio.keyslot = new radio_key
+	belt_radio.subspace_transmission = 1
+	belt_radio.canhear_range = 0
+	belt_radio.recalculateChannels()
+
+
+/obj/item/device/wormhole_lifebelt/Destroy()
+	SSobj.processing -= src
+	qdel(belt_radio)
+	. = ..()
+
+/obj/item/device/wormhole_lifebelt/process()
+	if(!medical_sensors)
+		return
+
+	var/mob/living/carbon/human/user
+	if(istype(loc, /mob/living/carbon/human))
+		user = loc
+	else
+		return
+
+	if(user.get_item_by_slot(slot_belt) != src)
+		return
+
+	if(user.InCritical())
+		if(ticks_in_crit == 0)
+			user << "<span class='notice'>You feel [src] growing warm and vibrating.</span>"
+		ticks_in_crit += 1
+		if(ticks_in_crit >= threshold)
+			feedback_add_details("jaunter", "M") // medical activation
+			belt_radio.talk_into(src, "Medical activation for [user] but yeah, go do whatever.", radio_channel)
+			activate(user)
+	else
+		ticks_in_crit = 0
+
+/obj/item/device/wormhole_lifebelt/attack_self(mob/user)
 	user.visible_message("<span class='notice'>[user.name] activates the [src.name]!</span>")
 	feedback_add_details("jaunter", "U") // user activated
 	activate(user)
 
-/obj/item/device/wormhole_jaunter/proc/turf_check(mob/user)
+/obj/item/device/wormhole_lifebelt/proc/turf_check(mob/user)
 	var/turf/device_turf = get_turf(user)
 	if(!device_turf||device_turf.z==2||device_turf.z>=7)
 		user << "<span class='notice'>You're having difficulties getting the [src.name] to work.</span>"
 		return FALSE
 	return TRUE
 
-/obj/item/device/wormhole_jaunter/proc/activate(mob/user)
-	if(!turf_check(user))
-		return
-
+/obj/item/device/wormhole_lifebelt/proc/get_beacons(mob/user)
+	// The shoes send you home, to where you belong
+	var/golem_mode = isgolem(user)
 	var/list/L = list()
+
 	for(var/obj/item/device/radio/beacon/B in world)
 		var/turf/T = get_turf(B)
-		if(T.z == ZLEVEL_STATION)
+		if(golem_mode)
+			if(istype(T.loc, /area/ruin/powered/golem_ship))
+				L += B
+		else if(T.z == ZLEVEL_STATION)
 			L += B
-	if(!L.len)
+
+	// If no beacons found for golems, home in on dat gold statue.
+	if(!L.len && golem_mode)
```

Comment:
This whole bit strikes me as snowflake and wrong also I still think the item is HUG BOX but I'll sleep on it